src/spec-node/configContainer.ts

```diff
@@ -71,26 +71,41 @@ async function resolveWithLocalFolder(params: DockerResolverParameters, parsedAu
 	return result;
 }
 
-export async function readDevContainerConfigFile(cliHost: CLIHost, workspace: Workspace | undefined, configFile: URI, mountWorkspaceGitRoot: boolean, output: Log, consistency?: BindMountConsistency, overrideConfigFile?: URI) {
+async function readFile(cliHost: CLIHost, path: URI) {
 	const documents = createDocuments(cliHost);
-	const content = await documents.readDocument(overrideConfigFile ?? configFile);
+	const content = await documents.readDocument(path);
 	if (!content) {
 		return undefined;
 	}
 	const raw = jsonc.parse(content) as DevContainerConfig | undefined;
 	const updated = raw && updateFromOldProperties(raw);
 	if (!updated || typeof updated !== 'object' || Array.isArray(updated)) {
-		throw new ContainerError({ description: `Dev container config (${uriToFsPath(configFile, cliHost.platform)}) must contain a JSON object literal.` });
+		throw new ContainerError({ description: `Dev container config (${uriToFsPath(path, cliHost.platform)}) must contain a JSON object literal.` });
 	}
-	const workspaceConfig = await getWorkspaceConfiguration(cliHost, workspace, updated, mountWorkspaceGitRoot, output, consistency);
+
+	return updated;
+}
+
+export async function readDevContainerConfigFile(cliHost: CLIHost, workspace: Workspace | undefined, configFile: URI, mountWorkspaceGitRoot: boolean, output: Log, consistency?: BindMountConsistency, overrideConfigFile?: URI) {
+	const confPath = overrideConfigFile ?? configFile;
+	let content = await readFile(cliHost, confPath) as DevContainerConfig;
+
+	if (content.extends) {
+		const extendsConfPath = path.resolve(path.dirname(confPath.path), content.extends);
+		const referencedContent = await readFile(cliHost, URI.file(extendsConfPath)) as DevContainerConfig;
```

Comment:
Should we check if there is an `extends` property on the referenced content?

Reply:
As-written, it would be present in the final configuration, but not resolved and it's contents merged in (it's easy enough to delete the property if you don't want it polluting the final configuration). I'm inclined to avoid support for recursive/multiple-extends unless we know there's a strong use-case for it. Detecting loops and self-references will make things far more complicated. Do you think it's important to support that immediately or can we wait?

Reply:
Waiting to assess the importance sounds good.